### PR TITLE
feat(behavior_path_planner): make initial ego pose in drivable area

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

When the initial ego pose is close to the boudary between two lanelets, the ego pose's footprint is sometimes out of the drivable area, which makes the trajectory points' velocities zero by obstacle_avoidance_planner, as shown in the following figure.

I expanded the drivable area properly so that the ego pose is inside the drivable area.

---
FYI
The same handling is already done for the goal pose (for end trajectory point)
https://github.com/autowarefoundation/autoware.universe/blob/ae8422b668441157e59bc72fa14bad46fd0ec8dd/planning/behavior_path_planner/src/utilities.cpp#L1140-L1143

So this PR means the same handling will be done for the initial pose (for start trajectory point)

## Related links
The second issue of https://github.com/autowarefoundation/autoware.universe/issues/641
image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/route_handler/src/route_handler.cpp [hunk 1]
```diff
@@ -353,22 +353,40 @@ lanelet::ConstPolygon3d RouteHandler::getIntersectionAreaById(const lanelet::Id
...
+
+  const double min_succeeding_length = vehicle_length * 2;
+  const auto succeeding_lanes_vec = lanelet::utils::query::getSucceedingLaneletSequences(
+    routing_graph_ptr_, goal_lanelet, min_succeeding_length);
+  if (succeeding_lanes_vec.empty()) {
+    return std::vector<lanelet::ConstLanelet>{};
+  }
+
+  return succeeding_lanes_vec.front();
 }
```
Comment (takayuki5168) [inline planning/route_handler/src/route_handler.cpp:390]: This is just refactoring without any behavior's change, in order to align with the new `getLanesBeforePose` function.

Comment (takayuki5168) [reply planning/route_handler/src/route_handler.cpp:390]: I'm not sure why `vehicle_length * 2` is used instead of just `vehicle_length`.

**Comment by TakaHoribe** [comment]
before
image

after
image

**Review by TakaHoribe — APPROVED** [review]
works fine good to go

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**